modules/unsupported/efeature/src/main/java/org/geotools/data/efeature/EFeatureFactoryFinder.java

```diff
@@ -80,19 +80,15 @@ public static EFeatureContextFactory getContextFactory()
             // Do the lousy slow system scan
             //
             synchronized (EFeatureFactoryFinder.class) {
```

Comment:
Repeating the if condition inside of the synchronized block would minimize the impact of the slow scan, particularly if the scan is so slow that other threads get stuck here while an iteration is executed.

Reply:
I'd rather not have my fingerprints on code that does concurrency _and_ interacts with GC if I didn't make an effort to properly understand it. It's also not related to what this PR is about.